sdx_lc/utils/db_utils.py

```diff
@@ -1,39 +1,51 @@
 import logging
 import os
+from urllib.parse import urlparse
 
 import pymongo
 
-DB_NAME = os.environ.get("DB_NAME")
-DB_CONFIG_TABLE_NAME = os.environ.get("DB_CONFIG_TABLE_NAME")
+
+def obfuscate_password_in_uri(uri: str) -> str:
+    """
+    Replace password field in URIs with a `*`, for logging.
+    """
+    parts = urlparse(uri)
```

Comment:
I'm running into a problem where `urlparse()` is failing to correctly parse the connection string when replica set is being used. Example:

```
MONGODB_CONNSTRING="mongodb://myDatabaseUser:D1fficultP%40ssw0rd @mongo1t:27027,mongo2t:27028,mongo3t:27029/?authSource=admin"
```

Then SDX-LC fails with the following error:
```
...
tenet-sdx-lc    |   File "/usr/src/app/sdx_lc/controllers/topology_controller.py", line 22, in <module>
tenet-sdx-lc    |     db_instance = DbUtils()
tenet-sdx-lc    |   File "/usr/src/app/sdx_lc/utils/db_utils.py", line 48, in __init__
tenet-sdx-lc    |     self.logger.info(f"[DB] Using {obfuscate_password_in_uri(mongo_connstring)}")
tenet-sdx-lc    |   File "/usr/src/app/sdx_lc/utils/db_utils.py", line 14, in obfuscate_password_in_uri
tenet-sdx-lc    |     return f"{parts.scheme}://{parts.username}:*@{parts.hostname}:{parts.port}/"
tenet-sdx-lc    |   File "/usr/local/lib/python3.9/urllib/parse.py", line 186, in port
tenet-sdx-lc    |     raise ValueError(message) from None
tenet-sdx-lc    | ValueError: Port could not be cast to integer value as '27027,mongo2t:27028,mongo3t:27029'
tenet-sdx-lc    | INFO:sdx_lc.app:SDX Local Controller starting up (name: None, domain: tenet.ac.za)
tenet-sdx-lc    | ERROR:connexion.apis.abstract:Failed to add operation for GET /SDX-LC/2.0.0/topology
```

Reply:
@sajith please check my suggestion right below if it makes sense to you. This simplifies the usage of the URI and avoid the issue reported above.

Reply:
Thanks for the suggestion Italo!